pyintegration/deephaven2/time.py

```diff
@@ -0,0 +1,775 @@
+    """
+    try:
+        return _JDateTimeUtils.currentTime()
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def date_at_midnight(dt, tz: TimeZone):
+    """ Returns a DateTime for the requested DateTime at midnight in the specified time zone.
+
+    Args:
+        dt (DateTime) - DateTime for which the new value at midnight should be calculated
+        tz: (TimeZone) - TimeZone for which the new value at midnight should be calculated
+
+    Returns:
+        DateTime
+
+    Raises:
+        DHError
+    """
+    try:
+
+        return _JDateTimeUtils.dateAtMidnight(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def day_of_month(dt, tz: TimeZone) -> int:
+    """ Returns an 1-based int value of the day of the month for a DateTime and specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the day of the month
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.dayOfMonth(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def day_of_week(dt, tz: TimeZone) -> int:
+    """ Returns an 1-based int value of the day of the week for a DateTime in the specified time zone, with 1 being
+     Monday and 7 being Sunday.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the day of the week.
+        tz (TimeZone): the TimeZone to use when interpreting the date/time.
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.dayOfWeek(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def day_of_year(dt, tz: TimeZone) -> int:
+    """ Returns an 1-based int value of the day of the year (Julian date) for a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the day of the year
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.dayOfYear(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def diff_nanos(dt1: DateTime, dt2: DateTime) -> int:
+    """ Returns the difference in nanoseconds between two DateTime values.
+
+    Args:
+        dt1 (DateTime): the 1st DateTime
+        dt2 (DateTime): the 2nd DateTime
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.diffNanos(dt1, dt2)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def format_datetime(dt, tz: TimeZone) -> str:
+    """ Returns a string date/time representation formatted as yyyy-MM-ddThh:mm:ss.nnnnnnnnn TZ.
+
+    Args:
+        dt (DateTime): the DateTime to format as a string
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
```

Comment:
I think you want `datetime` or `DateTime` and not `date/time`.  The last makes it look like a date or a time, while datetime variants are clear.  Check all of the docks for this.